### ISSUE TITLE
v5 component refactor: existing App Switcher compatibility (Top Bar)

### ISSUE THREAD

**Issue by nurm717123** [reapit/elements#253]
**Abstract**

_As part of the v5 Elements release, the ticket #153 was created. To break it into smaller part and allow for more focused discussions, this ticket has been created._

***app bar (parent)***
![image](https://github.com/user-attachments/assets/dbc50742-49e5-4c0d-aebf-2305aa72bfc8)

***this ticket***
![image](https://github.com/user-attachments/assets/abea3ff7-d037-4b0f-a890-8729a8de7980)

![image](https://github.com/user-attachments/assets/7c5e8c46-a585-48be-acd2-6f5306aa4dcc)

**Context**
- new app switcher not ready for dev yet
- while #222 will become the ticket for new AppSwitcher once the design is ready, this one is to use existing AppSwitcher in the new TopBar

As discussed in #153, this ticket involves creating a new app switcher based on the existing app switcher (`NavResponsiveAppSwitcher`) for the initial implementation.

**Specification**

- Is new component or existing? existing
- Is the component being re-named? No
- Is the component design signed off by product and design? not yet, but become part of App Bar / Top Bar ticket
- [Guide to component structure & documentation in storybook](https://github.com/reapit/elements/wiki/Developer-Contribution-Guide)
- [Link to DS in Figma](https://www.figma.com/design/XJ6qcAV8gHscsUodqJMNEF/Reapit-Elements-production-ready-components?node-id=16-4840&m=dev)
- [Link to Elements Audit in Figma](https://www.figma.com/design/XJ6qcAV8gHscsUodqJMNEF/Reapit-Elements-production-ready-components?node-id=16-4840&m=dev)
- [Accessibility requirements](https://m3.material.io/components/top-app-bar/accessibility)
- [Link to Design Guidelines](https://www.figma.com/design/pJ96bxoEFStjjIbAPy0Jaj/Design-Guidelines?node-id=945-25707&t=mvIlOCZjDYP90TZT-4)
- [Parent ticket](https://github.com/reapit/elements/issues/153)

**Developer Checklist**

- [ ] Styles alignment between Design System and Elements
- [ ] Check design tokens in Figma and implement CSS variable tokens if available for relevant component
- [ ] Align with accessibility standards / spec as per above
- [ ] If relevant, break down component into `Styles Only` and `React` component structures
- [ ] Ensure all variants of components are documented as appropriate 
- [ ] Ensure unit test coverage is adequate for component
- [ ] Update documentation in MDX file as per guidelines

**Release Checklist**

- [ ] Approved PR merged to main
- [ ] Design & product review and feedback addressed by developer
- [ ] Beta release by product / engineering lead to next beta version

**Additional Context or Information**

**Comment by adrian-reapit** [comment]
@nurm717123  Keeping the old AppSwitcher alive was mentioned to Andrei but can you check he realises it will be in the new AppBar

**Comment by kurtdoherty** [comment]
@nurm717123 Code snippet looks good, but just to confirm, is the margin/padding you mention the new `TopBar.AppSwitcher` handling going to be the margin/padding that we expect to use when the new AppSwitcher is implemented?

**Comment by nurm717123** [comment]
> [@nurm717123](https://github.com/nurm717123) Keeping the old AppSwitcher alive was mentioned to Andrei but can you check he realises it will be in the new AppBar

yes, sure 👍 

> [@nurm717123](https://github.com/nurm717123) Code snippet looks good, but just to confirm, is the margin/padding you mention the new `TopBar.AppSwitcher` handling going to be the margin/padding that we expect to use when the new AppSwitcher is implemented?

yes, it will

**Comment by kurtdoherty** [comment]
LGTM then 👍

**Comment by nurm717123** [comment]
> [@nurm717123](https://github.com/nurm717123) Keeping the old AppSwitcher alive was mentioned to Andrei but can you check he realises it will be in the new AppBar

I have confirmed this with him. We can use the existing AppSwitcher, as there are some app that currently utilizing it

**Comment by kurtdoherty** [comment]
FYI Console engineers will be taking on the development of the new App Switcher components in Elements 👍

**Comment by kurtdoherty** [comment]
Closing in favour of #428
